From 93da0c1f19a1bd1b00134519ba422a999e5b647a Mon Sep 17 00:00:00 2001
From: David Griffith <dave@661.org>
Date: Sun, 23 Apr 2017 10:31:17 -0700
Subject: [PATCH 28/68] Apply
 xv-20090826-paul-howarth-warnings-and-JP-FLmask.patch.mime.  See #4.

FLmask filters not implemented.  Still playing with that.
---
 xv.h       | 2 +-
 xvbutt.c   | 2 +-
 xvgif.c    | 6 +++---
 xvsunras.c | 2 +-
 xvwbmp.c   | 2 +-
 5 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/xv.h b/xv.h
index f5c7353..6eec214 100644
--- a/xv.h
+++ b/xv.h
@@ -1572,7 +1572,7 @@ int    CBTrack             PARM((CBUTT *));
 
 
 void   MBCreate            PARM((MBUTT *, Window, int, int, u_int, u_int,
-				 const char *, const char * const *, int,
+				 const char *, const char **, int,
 				 u_long, u_long, u_long, u_long));
 
 void   MBRedraw            PARM((MBUTT *));
diff --git a/xvbutt.c b/xvbutt.c
index b12cbc0..755b762 100644
--- a/xvbutt.c
+++ b/xvbutt.c
@@ -843,7 +843,7 @@ void MBCreate(mb, win, x, y, w, h, title, list, nlist, fg, bg, hi, lo)
      int           x,y;
      unsigned int  w,h;
      const char   *title;
-     const char  * const *list;
+     const char  **list;
      int           nlist;
      unsigned long fg,bg,hi,lo;
 {
diff --git a/xvgif.c b/xvgif.c
index 4266d41..89df98a 100644
--- a/xvgif.c
+++ b/xvgif.c
@@ -100,7 +100,7 @@ static void  doInterlace PARM((int));
 static int   gifError    PARM((PICINFO *, const char *));
 static void  gifWarning  PARM((const char *));
 
-static int         filesize;
+static long int   filesize;
 static const char *bname;
 static byte       *dataptr;
 
@@ -140,7 +140,7 @@ int LoadGIF(fname, pinfo)
   filesize = ftell(fp);
   fseek(fp, 0L, 0);
 
-  if (filesize + 256 < filesize)
+  if (filesize > (2147483647L - 256))
     return( gifError(pinfo, "GIF file size is too large") );
 
   /* the +256's are so we can read truncated GIF files without fear of
@@ -769,7 +769,7 @@ static int readImage(pinfo)
   pinfo->normw = pinfo->w;   pinfo->normh = pinfo->h;
 
   sprintf(pinfo->fullInfo,
-	  "GIF%s, %d bit%s per pixel, %sinterlaced.  (%d bytes)",
+	  "GIF%s, %d bit%s per pixel, %sinterlaced.  (%ld bytes)",
  	  (gif89) ? "89" : "87", BitsPerPixel,
 	  (BitsPerPixel==1) ? "" : "s",
  	  Interlace ? "" : "non-", filesize);
diff --git a/xvsunras.c b/xvsunras.c
index c533e89..e0ecc58 100644
--- a/xvsunras.c
+++ b/xvsunras.c
@@ -54,7 +54,7 @@ int LoadSunRas(fname, pinfo)
      PICINFO *pinfo;
 {
   FILE	*fp;
-  int	 linesize,lsize,csize,isize,i,w,h,d,npixels,nbytes;
+  unsigned int  linesize,lsize,csize,isize,i,w,h,d,npixels,nbytes;
   byte	 *image, *line;
   struct rasterfile sunheader;
   const char *bname;
diff --git a/xvwbmp.c b/xvwbmp.c
index 9dcb6c7..ae223d5 100644
--- a/xvwbmp.c
+++ b/xvwbmp.c
@@ -75,7 +75,7 @@ int LoadWBMP(fname, pinfo)
     int im_type;	/* image type (only type 0 supported) */
     uint8 fix_header;	/* fixed header field */
     int width, height;
-    int npixels, raw_size, aux;
+    unsigned int npixels, raw_size, aux;
     uint8 * raw;
 
     st_fname = fname;
-- 
2.17.1

